src/zenml/config/docker_settings.py

```diff
@@ -91,12 +91,19 @@ class DockerSettings(BaseSettings):
     --------------------------------
     Depending on the configuration of this object, requirements will be
     installed in the following order (each step optional):
-    - The packages installed in your local python environment
+    - The packages installed in your local python environment (extracted using
+      `pip freeze`)
     - The packages required by the stack unless this is disabled by setting
-      `install_stack_requirements=False`.
+      `install_stack_requirements=False`
     - The packages specified via the `required_integrations`
+    - The packages defined inside a pyproject.toml file given by the
```

Comment:
I think there is a misalignment with the docs here. In the docs, we say the priority is as follows:

1. Packages in the local env 
2. Packages for the stack
3. Required Integrations
4. Requirements (could be a list or a path to a requirements.txt)
5. Pyproject.toml next

In the docstring here, 4 and 5 are switched.

Reply:
I see that the docstring here is correct and the docs need to be updated.

Reply:
You're right, I fixed the error in the docs.

In general, there are two scenarios:
- You don't explicitly specify any requirements using docker settings: In this case we first look for a `requirements.txt` (-> highest priority), and if it doesn't exist we look for a `pyproject.toml`
- You do explicitly specify some requirements (using e.g. `DockerSettings.required_integrations` or `DockerSettings.requirements`): In this case the order in the docstring is the one that is being used. Requirements being last means it is the highest priority (it will override values from all previous steps), which is in line with the first case.

Do you think these make sense?

Reply:
I left a comment about the confusion that I had with the interaction between the implicit and explicit definition of requirements but outside of that, the explanation looks good here.